{{cookiecutter.repository_name}}/README.md

```diff
@@ -1,10 +1,116 @@
 [![Travis CI Shield](https://travis-ci.org/{{ cookiecutter.github_owner }}/{{ cookiecutter.repository_name }}.svg?branch=master)](https://travis-ci.org/{{ cookiecutter.github_owner }}/{{ cookiecutter.repository_name }})
+{%- if cookiecutter.use_codecov == 'y' %}
+[![Coverage Status](https://codecov.io/gh/{{ cookiecutter.github_owner }}/{{ cookiecutter.repository_name }}/branch/master/graph/badge.svg)](https://codecov.io/gh/{{ cookiecutter.github_owner }}/{{ cookiecutter.repository_name }})
+{%- endif %}
+[![Downloads](https://pepy.tech/badge/{{ cookiecutter.package_name }})](https://pepy.tech/project/{{ cookiecutter.package_name }})
 
 # {{ cookiecutter.project_name }}
 
 {{ cookiecutter.project_short_description }}
 
+{% if cookiecutter.open_source_license == 'Not open source' -%}
 - Free software: {{ cookiecutter.open_source_license }}
+{% endif -%}
 - Documentation: https://{{ cookiecutter.github_owner }}.github.io/{{ cookiecutter.repository_name }}
 - Homepage: https://github.com/{{ cookiecutter.github_owner }}/{{ cookiecutter.repository_name }}
+
+# Overview
+
+TODO: Provide a short overview of the project here.
+
+# Install
+
+## Requirements
+
+**{{ cookiecutter.project_name }}** has been developed and tested on [Python {%- if cookiecutter.support_py2 == 'y' %}2.7, {%- endif %}3.4, 3.5, 3.6 and 3.7](https://www.python.org/downloads/)
+
+Also, although it is not strictly required, the usage of a [virtualenv](https://virtualenv.pypa.io/en/latest/)
+is highly recommended in order to avoid interfering with other software installed in the system
+where **{{ cookiecutter.project_name }}** is run.
+
+These are the minimum commands needed to create a virtualenv using python3.6 for **{{ cookiecutter.project_name }}**:
+
+```bash
+pip install virtualenv
+virtualenv -p $(which python3.6) {{ cookiecutter.repository_name }}-venv
+```
+
+Afterwards, you have to execute this command to have the virtualenv activated:
+
+```bash
+source {{ cookiecutter.repository_name }}-venv/bin/activate
+```
+
+Remember about executing it every time you start a new console to work on **{{ cookiecutter.project_name }}**!
+
+## Install with pip
+
+After creating the virtualenv and activating it, we recommend using
+[pip](https://pip.pypa.io/en/stable/) in order to install **{{ cookiecutter.project_name }}**:
+
+```bash
+pip install {{ cookiecutter.package_name }}
+```
+
+This will pull and install the latest stable release from [PyPi](https://pypi.org/).
+
+## Install from source
+
+Alternatively, with your virtualenv activated, you can clone the repository and install it from
+source by running `make install` on the `stable` branch:
+
+```bash
+git clone git@github.com:{{ cookiecutter.github_owner }}/{{ cookiecutter.repository_name }}.git
+cd {{ cookiecutter.repository_name }}
+git checkout stable
+make install
+```
+
+## Install for Development
```

Comment:
Done!

But there is something that we need to think about: when setting the link, we either point at the CONTRIBUTING.rst file in github, or point at the contributing.html site in the documentation.

If we point at the first one, it will always work when clicking from the README, but the link will be broken on Sphinx.
If we point at the second one, the link will be broken on the README until the documentation exists.

On way to solve this would be to figure out a way to replace the relative links, so that github points to github, but sphinx points at sphinx.

Reply:
yes let's see if we can figure out relative links